staging/src/k8s.io/api/resource/v1alpha2/types.go

```diff
@@ -462,3 +523,208 @@ type ResourceClaimTemplateList struct {
+
+// ResourceClaimParameters defines resource requests for a ResourceClaim in an
+// in-tree format understood by Kubernetes.
+type ResourceClaimParameters struct {
+	metav1.TypeMeta `json:",inline"`
+	// Standard object metadata
+	metav1.ObjectMeta `json:"metadata,omitempty" protobuf:"bytes,1,opt,name=metadata"`
+
+	// If this object was created from some other resource, then this links
+	// back to that resource. This field is used to find the in-tree representation
+	// of the claim parameters when the parameter reference of the claim refers
+	// to some unknown type.
+	// +optional
+	GeneratedFrom *ResourceClaimParametersReference `json:"generatedFrom,omitempty" protobuf:"bytes,2,opt,name=generatedFrom"`
+
+	// Shareable indicates whether the allocated claim is meant to be shareable
+	// by multiple consumers at the same time.
+	// +optional
+	Shareable bool `json:"shareable,omitempty" protobuf:"bytes,3,opt,name=shareable"`
+
+	// DriverRequests describes all resources that are needed for the
+	// allocated claim. A single claim may use resources coming from
+	// different drivers. For each driver, this array has at most one
+	// entry which then may have one or more per-driver requests.
+	//
+	// May be empty, in which case the claim can always be allocated.
+	//
+	// +listType=atomic
+	DriverRequests []DriverRequests `json:"driverRequests,omitempty" protobuf:"bytes,4,opt,name=driverRequests"`
+}
+
+// DriverRequests describes all resources that are needed from one particular driver.
+type DriverRequests struct {
+	// DriverName is the name used by the DRA driver kubelet plugin.
+	DriverName string `json:"driverName,omitempty" protobuf:"bytes,1,opt,name=driverName"`
+
+	// VendorParameters are arbitrary setup parameters for all requests of the
+	// claim. They are ignored while allocating the claim.
+	//
+	// +optional
+	VendorParameters runtime.RawExtension `json:"vendorParameters,omitempty" protobuf:"bytes,2,opt,name=vendorParameters"`
+
+	// Requests describes all resources that are needed from the driver.
+	// +listType=atomic
+	Requests []ResourceRequest `json:"requests,omitempty" protobuf:"bytes,3,opt,name=requests"`
+}
+
+// ResourceRequest is a request for resources from one particular driver.
+type ResourceRequest struct {
+	// VendorParameters are arbitrary setup parameters for the requested
+	// resource. They are ignored while allocating a claim.
+	//
+	// +optional
+	VendorParameters runtime.RawExtension `json:"vendorParameters,omitempty" protobuf:"bytes,1,opt,name=vendorParameters"`
+
+	ResourceRequestModel `json:",inline" protobuf:"bytes,2,name=resourceRequestModel"`
+}
+
+// ResourceRequestModel must have one and only one field set.
+type ResourceRequestModel struct {
+	// NamedResources describes a request for resources with the named resources model.
+	//
+	// +optional
+	NamedResources *NamedResourcesRequest `json:"namedResources,omitempty" protobuf:"bytes,1,opt,name=namedResources"`
```

Comment:
I'm less optimistic that we can make "named resources" *the one-and-only* model that Kubernetes will ever need and have. If we drop the one-of-many structs, we close the door for adding another model because clients cannot detect that they are not handling some new, unknown model.

Reply:
I know we're not quite eye-to-eye here yet.